framework/src/media/MediaRecorderImpl.cpp

```diff
@@ -582,27 +591,35 @@ void MediaRecorderImpl::notifySync()
 	mSyncCv.notify_one();
 }
 
-void MediaRecorderImpl::notifyObserver(observer_command_t cmd)
+void MediaRecorderImpl::notifyObserver(observer_command_t cmd, recorder_result_t errCode)
 {
 	medvdbg("notifyObserver cmd : %d\n", cmd);
 	if (mRecorderObserver) {
 		RecorderObserverWorker& row = RecorderObserverWorker::getWorker();
 		switch (cmd) {
 		case OBSERVER_COMMAND_STARTED: {
 			medvdbg("OBSERVER_COMMAND_STARTED\n");
-			row.enQueue(&MediaRecorderObserverInterface::onRecordStarted, mRecorderObserver, mId);
+			row.enQueue(&MediaRecorderObserverInterface::onRecordStarted, mRecorderObserver, mRecorder);
 		} break;
 		case OBSERVER_COMMAND_PAUSED: {
 			medvdbg("OBSERVER_COMMAND_PAUSED\n");
-			row.enQueue(&MediaRecorderObserverInterface::onRecordPaused, mRecorderObserver, mId);
+			row.enQueue(&MediaRecorderObserverInterface::onRecordPaused, mRecorderObserver, mRecorder);
 		} break;
 		case OBSERVER_COMMAND_FINISHIED: {
 			medvdbg("OBSERVER_COMMAND_FINISHIED\n");
-			row.enQueue(&MediaRecorderObserverInterface::onRecordFinished, mRecorderObserver, mId);
+			row.enQueue(&MediaRecorderObserverInterface::onRecordFinished, mRecorderObserver, mRecorder);
+		} break;
+		case OBSERVER_COMMAND_START_ERROR: {
```

Comment:
I think OBSERVER_COMMAND_ERROR is enough let's distinguish later, if it needed later.
I mean more in detail about which function occurs, I think error log is enough about fucntion info, because application doesn't have to make each of handling logic.

Reply:
I think it's a good idea to divide each of these to handle the media object when the callback function is called in the app.

OnRecordError When used as a single function, we need another parameter that tells us when it happened.